server/api/rounds.py

```diff
@@ -518,6 +519,41 @@ def sample_ballots(
     contests: List[Contest],
     sample_sizes: Dict[str, int],
 ):
+    participating_jurisdictions = {
+        jurisdiction for contest in contests for jurisdiction in contest.jurisdictions
+    }
+
+    # Audits must be deterministic and repeatable for the same real world
+    # inputs. So the sampler expects the same input for the same real world
+    # data. Thus, we use the jurisdiction name and batch keys
+    # (deterministic real world ids) instead of the jurisdiction and batch
+    # ids (non-deterministic uuids that we generate for each audit).
+    batch_id_to_key = {
+        batch.id: (jurisdiction.name, batch.tabulator, batch.name)
+        for jurisdiction in participating_jurisdictions
+        for batch in jurisdiction.batches
+    }
+    batch_key_to_id = {
```

Comment:
Is there not a builtin that does this for you? Seems a little clunky, but I'm not sure.

Reply:
not that i could find